merlin/config/merlin_config_manager.py

```diff
@@ -0,0 +1,214 @@
+        if self.args.type == "redis":
+            self.update_redis_config(broker_config)
+        elif self.args.type == "rabbitmq":
+            self.update_rabbitmq_config(broker_config)
+        else:
+            LOG.error("Invalid broker type. Use 'redis' or 'rabbitmq'.")
+            return
+
+        config["broker"] = broker_config
+
+        with open(self.config_file, "w") as app_yaml_file:
+            yaml.dump(config, app_yaml_file, default_flow_style=False)
+
+        LOG.info("Broker settings successfully updated. Check your new connection with `merlin info`.")
+
+    def update_backend(self):
+        """
+        Update the results backend section of the app.yaml file.
+        """
+        LOG.info(f"Updating results backend settings in '{self.config_file}'...")
+
+        with open(self.config_file, "r") as app_yaml_file:
+            config = yaml.safe_load(app_yaml_file)
+
+        backend_config = config.get("results_backend", {})
+        if self.args.type == "redis":
+            self.update_redis_config(backend_config)
+        else:
+            LOG.error("Invalid backend type. Use 'redis'.")
+            return
+
+        config["results_backend"] = backend_config
+
+        with open(self.config_file, "w") as app_yaml_file:
+            yaml.dump(config, app_yaml_file, default_flow_style=False)
+
+        LOG.info("Results backend settings successfully updated. Check your new connection with `merlin info`.")
+
+    def update_redis_config(self, config: Dict[str, Union[str, int]]):
+        """
+        Update the Redis-specific configuration for either broker or backend.
+
+        Args:
+            config: The configuration dictionary to update.
+        """
+        LOG.warning("Redis does not use the 'username' or 'vhost' arguments. Ignoring these if provided.")
+        config["name"] = "rediss"
+        config["username"] = ""  # Redis doesn't use a username
+
+        required_fields = ["password_file", "server", "port", "db_num", "cert_reqs"]
+        self.update_config(config, required_fields)
+
+    def update_rabbitmq_config(self, config: Dict[str, Union[str, int]]):
+        """
+        Update the RabbitMQ-specific broker configuration.
+
+        Args:
+            config: The configuration dictionary to update.
+        """
+        LOG.warning("RabbitMQ does not use the 'db_num' argument.")
+        config["name"] = "rabbitmq"
+
+        required_fields = ["username", "password_file", "server", "port", "vhost", "cert_reqs"]
+        self.update_config(config, required_fields)
+
+    def update_config(self, config: Dict[str, Union[str, int]], required_fields: List[str]):
+        """
+        Generic function to update configuration fields based on provided arguments.
+
+        Args:
+            config: The configuration dictionary to update.
+            required_fields: List of required field names for the configuration.
+        """
+        for field in required_fields:
```

Comment:
Anyway to check the old value And then add a LOG info if new value is different?